### PR TITLE
Buy now button

### PR DESCRIPTION
**Pull request by irismoini**
- wired buy now button to add order items & orders to database

### CODE REVIEW

File: app/javascript/components/EditItemForm.jsx [hunk 1]
```diff
@@ -67,7 +68,9 @@ export default (props) => {
             return "Doughnut quantity is missing or invalid"
         } else if (typeof doughnut?.description !== "string") {
             return "Doughnut description is invalid";
-        }
+        } else if (typeof doughnut?.image_url !== "string" || doughnut.image_url.length <= 0) {
+            return "Doughnut image url is invalid"
+        } //TODO: include additional verification to see if this is a valid path
```
Comment (andrewhlu) [inline app/javascript/components/EditItemForm.jsx]: I don't think we need to force sellers to provide an image URL with every new doughnut. We can always default to the existing doughnut images if they don't provide one.

File: app/javascript/components/ItemCard.jsx [hunk 2]
```diff
@@ -3,9 +3,57 @@ import Placeholder1 from "images/item_placeholder_1.JPG";
...
+	const [cookies, _setCookie, _removeCookie] = useCookies(["ddough-auth"]);
+	const [userId] = useState();
+	
+
+	const buyNowHandler = async (e) => {
+        e.preventDefault();
+        setErrorMessage(null);
+
+        const body = {
+			doughnut_id: props.idx,
```
Comment (andrewhlu) [inline app/javascript/components/ItemCard.jsx]: `doughnut_id` and `idx` are not the same - the former is the doughnut's ID in the database and the latter is the position of the doughnut in the array of items that is stored in the frontend upon making the API call to get doughnuts. You'll want to pass down the doughnut ID from the item object and pass that in instead.

File: app/javascript/components/ItemCard.jsx [hunk 3]
```diff
@@ -3,9 +3,57 @@ import Placeholder1 from "images/item_placeholder_1.JPG";
 import Placeholder2 from "images/item_placeholder_2.JPG";
 import EditIcon from "images/icons/edit_icon.svg";
 import DeleteIcon from "images/icons/delete_icon.svg";
+import { useCookies } from "react-cookie";
+
 
 export default(props) => {
 	const [onHover, setOnHover] = useState(false);
+	const [errorMessage, setErrorMessage] = useState(null);
```
Comment (andrewhlu) [inline app/javascript/components/ItemCard.jsx:11]: Do we have plans to use / display this error message?

**Review by andrewhlu — CHANGES_REQUESTED** [review]
In addition to the comments below, it might be a good idea to PR the image URL and buy now changes separately.

File: app/javascript/components/ItemCard.jsx (hunk 3 above)
Comment (irismoini) [reply app/javascript/components/ItemCard.jsx:11]: I don't think we do, it's just something I included incase we did want to/for consistency sake.

**Comment by irismoini** [comment]
> In addition to the comments below, it might be a good idea to PR the image URL and buy now changes separately.

Yeah I realized I had checkout the buy now branch while on the image url branch, but I did a bunch rebasing to fix that. Should look good now.

**Pull request merged**